cpp/src/rolling/detail/rolling.cuh

```diff
@@ -126,9 +126,16 @@ struct DeviceRolling {
 
     bool output_is_valid = (count >= min_periods);
 
-    // store the output value, one per thread
-    cudf::detail::rolling_store_output_functor<OutputType, op == aggregation::MEAN>{}(
-      output.element<OutputType>(current_index), val, count);
+    if (output_is_valid) {
+      // store the output value, one per thread, but only if the
+      // output is valid. min_periods is required to be >= 1, and so
+      // here, count must be nonzero. We need to avoid storing if
+      // count is zero since this could could UB in some aggregations,
```

Comment:
```suggestion
      // count is zero since this could UB in some aggregations,
```

Reply:
I think this is what was meant:
```suggestion
      // count is zero since this could cause UB in some aggregations,
```

Reply:
I confess that I've used 'UB' as a verb in the past.